bliss/simulator/prior.py

```diff
@@ -209,33 +207,38 @@ def _sample_n_galaxies_and_stars(self, n_sources):
 
     def _sample_star_fluxes(self, star_ratios):
         latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
-        fluxes = torch.zeros(
-            self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands
+        r_flux = self._draw_truncated_pareto(
+            self.galaxy_alpha, self.galaxy_flux_min, self.galaxy_flux_max, latent_dims
         )
-        # sample value for first band from pareto for each light source
-        fluxes[:, :, :, :, 0] = self._draw_truncated_pareto(
-            self.star_flux_alpha, self.star_flux_min, self.star_flux_max, latent_dims
+        total_flux = self.multiflux(star_ratios, r_flux)
+
+        # select specified bands
+        select_flux = torch.zeros(
+            self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, self.n_bands
         )
-        # Compute remaining band fluxes via ratio from SDSS catalog and sampled values
-        if self.n_bands > 1:
-            fluxes[:, :, :, :, 1:] = self.multiflux(star_ratios, fluxes[:, :, :, :, 0])
-        return fluxes
+        for i, bnd in enumerate(self.bands):
+            select_flux[:, :, :, :, i] = total_flux[:, :, :, :, bnd]
```

Comment:
@sawanp813  instead of a for loop, can we just index by self.bands? Then instead of these lines, all we need is `select_flux = total_flux[..., self.bands]` (we may need to convert self.bands to a numpy array first)

Reply:
Yeah the for-loop wasn't ideal, and I didn't know that was possible - will do!